### DISCUSSION TITLE
Filter media files by types in picker modal

### DISCUSSION THREAD

**Discussion by lotarbo** [awcodes/filament-curator#223] [Ideas]
Hello, in a project i  have many different files, and i think, if we can filter files in media picker popup by **acceptedFileTypes** in MediaController

```php
->when($request->has('types'), function($query) use ($request) {
                return $query->whereIn('type', $request->types);
            })
```

 we can increase performance and usability)
What do you think?

**Comment by lotarbo** [comment]
also i think we can solve the issue with non-strict types)
`curator/media?types[]=video/*&types[]=image/*`

very dirty code!)
```php
->when($request->has('types'), function($query) use ($request) {
                $query = $query->whereIn('type', $request->types);
                $globalTypes = collect($request->types)->filter(fn($type) =>str_contains($type, '*'));
                if ($globalTypes) {
                    $globalTypes->map(fn($type) => $query->orWhere('type', 'LIKE', str_replace('*', '', $type) . '%'));
                }
                return $query;
            })
```

**Comment by awcodes** [comment]
Thanks, I'll be able to look at this more tomorrow. Will keep you posted.

**Comment by awcodes** [comment]
v2.8.0 released. Let me know if this fixes all your issues.
